vm/src/serde/deserialize_program.rs

```diff
@@ -349,6 +352,22 @@ pub fn deserialize_felt_hex<'de, D: Deserializer<'de>>(d: D) -> Result<Felt252,
     d.deserialize_str(Felt252Visitor)
 }
 
+pub fn serialize_program_data<S: Serializer>(
+    v: &[MaybeRelocatable],
+    serializer: S,
+) -> Result<S::Ok, S::Error> {
+    let v = v
+        .iter()
+        .map(|val| match val.clone() {
+            MaybeRelocatable::Int(value) => format!("0x{:x}", value.to_biguint()),
```

Comment:
We could leave this as is for now, but this hints to a need to implement `Display` for `Felt252`. @fmoletta WDYT?

Reply:
There is a `Display` implementation for `Felt252`

Reply:
I guess it's done this way because the `Display`/`Debug` impls produce a decimal representation instead of a hexadecimal one. This use case is equivalent to `value.to_str_radix(16)`

Reply:
Doesn't the `Display` trait receive a formatter? Or is that a different trait? Typically you'll get one and then the format string can be processed to pick the right representation.

Reply:
Since @MegaRedHand created the issue at lambdaworks I'll mark this one resolved.